src/components/static-map.js

```diff
@@ -192,51 +207,59 @@ export default class StaticMap extends PureComponent {
     return null;
   }
 
+  _renderOverlays(dimensions) {
+    const {
+      width = this.props.width,
+      height = this.props.height
+    } = dimensions;
+    this._updateMapSize(width, height);
+
+    const staticContext = {
+      viewport: new WebMercatorViewport(Object.assign({}, this.props, {
+        width,
+        height
+      }))
+    };
+
+    return createElement(StaticContext.Provider, {value: staticContext},
+      createElement('div', {
+        key: 'map-overlays',
+        className: 'overlays',
+        style: CONTAINER_STYLE,
+        children: this.props.children
+      })
+    );
+  }
+
   render() {
     const {className, width, height, style, visibilityConstraints} = this.props;
-    const mapContainerStyle = Object.assign({}, style, {width, height, position: 'relative'});
+    const mapContainerStyle = Object.assign({position: 'relative'}, style, {width, height});
 
     const visible = this.props.visible &&
       checkVisibilityConstraints(this.props.viewState || this.props, visibilityConstraints);
 
-    const mapStyle = Object.assign({}, style, {
-      width,
-      height,
+    const mapStyle = Object.assign({}, CONTAINER_STYLE, {
       visibility: visible ? 'visible' : 'hidden'
     });
-    const overlayContainerStyle = {
-      position: 'absolute',
-      left: 0,
-      top: 0,
-      width,
-      height,
-      overflow: 'hidden'
-    };
-    const staticContext = {
-      viewport: new WebMercatorViewport(this.props)
-    };
 
-    return createElement(StaticContext.Provider, {value: staticContext},
-      createElement('div', {
-        key: 'map-container',
-        style: mapContainerStyle,
-        children: [
-          createElement('div', {
-            key: 'map-mapbox',
-            ref: this._mapboxMapLoaded,
-            style: mapStyle,
-            className
-          }),
-          createElement('div', {
-            key: 'map-overlays',
-            className: 'overlays',
-            style: overlayContainerStyle,
-            children: this.props.children
-          }),
-          this._renderNoTokenWarning()
-        ]
-      })
-    );
+    return createElement('div', {
+      key: 'map-container',
+      style: mapContainerStyle,
+      children: [
+        createElement('div', {
+          key: 'map-mapbox',
+          ref: this._mapboxMapLoaded,
+          style: mapStyle,
+          className
+        }),
+        createElement(AutoSizer, {
+          key: 'autosizer',
+          disableWidth: Number.isFinite(width),
+          disableHeight: Number.isFinite(height)
+        }, this._renderOverlays.bind(this)),
```

Comment:
I would do this in constructor or just use class properties.

Reply:
This is subtle... We need `AutoSizer` to rerender when either the props of `StaticMap` change or the container is resized. But it's a pure component. `bind` makes a new child every render pass and triggers that rerender.